packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -772,6 +772,23 @@ CodeGuru profiling is supported for all Java runtimes and Python3.6+ runtimes.
 See [the AWS documentation](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/setting-up-lambda.html)
 to learn more about AWS Lambda's Profiling support.
 
+## Lambda with SnapStart
+
+SnapStart is currently supported only on Java 11 runtime. After you enable Lambda SnapStart for a particular Lambda function, publishing a new version of the function will trigger an optimization process.
+
+See [the AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html) to learn more about AWS Lambda SnapStart
+
+```ts
+const fn = new lambda.Function(this, 'MyFunction', {
+  code: lambda.Code.fromAsset('handler.zip'),
+  runtime: lambda.Runtime.JAVA_11,
+  handler: 'example.Handler::handleRequest',
+  snapStart: true,
```

Comment:
I concur with @TheRealAmazonKendra. There's too many restrictions on the lambda to put this as a constructor prop (from https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html):

>SnapStart supports the Java 11 [runtime](https://docs.aws.amazon.com/lambda/latest/dg/lambda-java.html).

>SnapStart does not support [provisioned concurrency](https://docs.aws.amazon.com/lambda/latest/dg/provisioned-concurrency.html), the [arm64 architecture](https://docs.aws.amazon.com/lambda/latest/dg/foundation-arch.html), the [Lambda Extensions API](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html), [Amazon Elastic File System (Amazon EFS)](https://docs.aws.amazon.com/lambda/latest/dg/services-efs.html), [AWS X-Ray](https://docs.aws.amazon.com/lambda/latest/dg/services-xray.html), or ephemeral storage greater than 512 MB.

it's good that you added error checking for the Java11 runtime case, but all of these other restrictions must also be checked at synth time. This bloats the constructor with unnecessary checks that will eventually be removed when Lambda rolls this out to more function types. 

Because we need those type checks today, and we will one day have to deprecate them, I agree with @TheRealAmazonKendra's suggestion to make this a new method (maybe `enableSnapStart()`?) on the `Function` class that will do all of this error checking in one place.